lang/src/org/partiql/lang/eval/CompileOptions.kt

```diff
@@ -79,6 +79,7 @@ enum class TypingMode {
  * Indicates how CAST should behave.
  */
 enum class TypedOpBehavior {
+    @Deprecated("Please use HONOR_PARAMETERS")
```

Comment:
May be helpful to include the other fields of the `@Deprecated` annotation. See [@Deprecated](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-deprecated/)'s docs.

As a suggestion could use `message = "TypedOpBehavior.LEGACY is an old compile option that will be removed in an upcoming release"` and `replaceWith = "TypedOpBehavior.HONOR_PARAMETERS"`.